src/transformers/models/maskformer/modeling_maskformer.py

```diff
@@ -0,0 +1,2502 @@
+        matcher: MaskFormerHungarianMatcher,
+        weight_dict: Dict[str, float],
+        eos_coef: float,
+    ):
+        """
+        The MaskFormer Loss. The loss is computed very similar to DETR. The process happens in two steps: 1) we compute
+        hungarian assignment between ground truth masks and the outputs of the model 2) we supervise each pair of
+        matched ground-truth / prediction (supervise class and mask)
+
+        Args:
+            num_classes (`int`):
+                The number of classes.
+            matcher ([`MaskFormerHungarianMatcher]`):
+                A torch module that computes the assigments between the predictions and labels.
+            weight_dict (`Dict[str, float]`):
+                A dictionary of weights to be applied to the different losses.
+            eos_coef (float):
+                Weight to apply to the null class.
+        """
+
+        super().__init__()
+        requires_backends(self, ["scipy"])
+        self.num_classes = num_classes
+        self.matcher = matcher
+        self.weight_dict = weight_dict
+        self.eos_coef = eos_coef
+        empty_weight = torch.ones(self.num_classes + 1)
+        empty_weight[-1] = self.eos_coef
+        self.register_buffer("empty_weight", empty_weight)
+
+    def loss_labels(
+        self, class_queries_logits: Tensor, class_labels: Tensor, indices: Tuple[np.array]
+    ) -> Dict[str, Tensor]:
+        """Compute the losses related to the labels using cross entropy.
+
+        Args:
+            class_queries_logits (`torch.Tensor`):
+                A tensor of shape `batch_size, num_queries, num_classes`
+            class_labels (`Dict[str, Tensor]`):
+                A tensor of shape `batch_size, num_classes`
+            indices (`Tuple[np.array])`:
+                The indices computed by the Hungarian matcher.
+
+        Returns:
+            `Dict[str, Tensor]`: A dict of `torch.Tensor` containing the following key:
+            - **loss_cross_entropy** -- The loss computed using cross entropy on the predicted and ground truth labels.
+        """
+
+        pred_logits = class_queries_logits
+        batch_size, num_queries, _ = pred_logits.shape
+
+        idx = self._get_predictions_permutation_indices(indices)
+        # shape = [BATCH, N_QUERIES]
+        target_classes_o = torch.cat([target[j] for target, (_, j) in zip(class_labels, indices)])
+        # shape = [BATCH, N_QUERIES]
+        target_classes = torch.full(
+            (batch_size, num_queries), fill_value=self.num_classes, dtype=torch.int64, device=pred_logits.device
+        )
+        target_classes[idx] = target_classes_o
+        # target_classes is a [BATCH, CLASSES, N_QUERIES], we need to permute pred_logits "b q c -> b c q"
+        pred_logits_permuted = pred_logits.permute(0, 2, 1)
+        loss_ce = cross_entropy(pred_logits_permuted, target_classes, self.empty_weight)
+        losses: Tensor = {"loss_cross_entropy": loss_ce}
```

Comment:
```suggestion
        losses = {"loss_cross_entropy": loss_ce}
```
The type hint is not accurate.